internal/dag/policy.go

```diff
@@ -315,6 +315,9 @@ func escapeHeaderValue(value string, dynamicHeaders map[string]string) string {
 	// REQ(header-name)
 	var validReqEnvoyVar = regexp.MustCompile(`%(%REQ\(:?[\w-]+(\?:?[\w-]+)?\)(:\d+)?%)%`)
 	escapedValue = validReqEnvoyVar.ReplaceAllString(escapedValue, "$1")
+	// RESP(header-name) similar to REQ
+	var validRespEnvoyVar = regexp.MustCompile(`%(%RESP\(:?[\w-]+(\?:?[\w-]+)?\)(:\d+)?%)%`)
```

Comment:
@hmouaad I would think we'd only want to allow this in a ResponseHeadersPolicy? I'm not sure off the top of my head what Envoy's behavior would be if this was specified in `request_headers_to_add`, but seems like good validation for Contour to just not allow it. That will require some further refactoring here.